src/Graphics/Vty/Attributes/Color.hs

```diff
@@ -119,9 +157,40 @@ brightMagenta= ISOColor 13
 brightCyan   = ISOColor 14
 brightWhite  = ISOColor 15
 
+-- TODO: test this with other terminals
+linearColor :: Integral i => i -> i -> i -> Color
+linearColor r g b
+    | r < 0 || r > 255 = makeError
+    | g < 0 || g > 255 = makeError
+    | b < 0 || b > 255 = makeError
+    | otherwise = RGBColor r' g' b'
+    where
+        r' = fromIntegral r :: Word8
+        g' = fromIntegral g :: Word8
+        b' = fromIntegral b :: Word8
+        makeError = error $ printf "invalid value in rgbColor: (%d, %d, %d)" r' g' b'
```

Comment:
I don't know if you have the time to make a change to this PR -- I know it's been a while since you wrote it -- but if so, I think it would be better if this clamped color values to the [0..255] range rather than raising an exception.

Reply:
Yeah, it is better. Fixed.